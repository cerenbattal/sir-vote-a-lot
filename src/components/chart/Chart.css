.chart {
    background-color: #F7F7F7; 
    text-align: center;
    border-radius: 5px;
    padding: 1em;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    min-height: 600px;
}

.footer > p {
    float: left;
    margin-left: 0.5em;
}