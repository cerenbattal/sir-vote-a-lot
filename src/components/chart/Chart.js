import React from 'react'
import SimpleBarChart from './SimpleBarChart'
import './Chart.css'

export default function Chart() {
    return (
        <div data-testid="chart-test" className="chart">
            <SimpleBarChart data-testid="barchart-test"/>
            <div className="footer">
                <p>Total votes: 1</p>
            </div>
        </div>
    )
}
