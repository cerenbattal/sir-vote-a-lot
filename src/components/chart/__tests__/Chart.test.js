import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import Chart from '../Chart'

afterEach(() => {
    cleanup();
})

test('should render chart component', () => {
    render(<Chart />);
    const chartElement = screen.getByTestId('chart-test');
    expect(chartElement).toBeInTheDocument();
})

test('matches snapshot', () => {
    const tree = renderer.create(<Chart />);
    expect(tree).toMatchSnapshot();
})