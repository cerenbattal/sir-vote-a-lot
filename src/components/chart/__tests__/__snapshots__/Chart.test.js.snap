// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="chart"
  data-testid="chart-test"
>
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": 300,
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "99%",
      }
    }
  />
  <div
    className="footer"
  >
    <p>
      Total votes: 1
    </p>
  </div>
</div>
`;
